assemble/bin/accumulo

```diff
@@ -51,137 +75,179 @@ locationByProgram()
-esac
-
-XML_FILES="${ACCUMULO_CONF_DIR}"
-LOG4J_JAR=$(find -H "${HADOOP_PREFIX}/lib" "${HADOOP_PREFIX}"/share/hadoop/common/lib -name 'log4j*.jar' -print 2>/dev/null | head -1)
-SLF4J_JARS="${ACCUMULO_HOME}/lib/slf4j-api.jar:${ACCUMULO_HOME}/lib/slf4j-log4j12.jar"
-
-# The `find` command could fail for environmental reasons or bad configuration
-# Avoid trying to run Accumulo when we can't find the jar
-if [ -z "${LOG4J_JAR}" -a -z "${CLASSPATH}" ]; then
-   echo "Could not locate Log4j jar in Hadoop installation at ${HADOOP_PREFIX}"
-   exit 1
-fi
-
-CLASSPATH="${XML_FILES}:${START_JAR}:${SLF4J_JARS}:${LOG4J_JAR}:${CLASSPATH}"
-
-if [ -z "${JAVA_HOME}" -o ! -d "${JAVA_HOME}" ]; then
-   echo "JAVA_HOME is not set or is not a directory.  Please make sure it's set globally or in conf/accumulo-env.sh"
-   exit 1
-fi
-if [ -z "${HADOOP_PREFIX}" -o ! -d "${HADOOP_PREFIX}" ]; then
-   echo "HADOOP_PREFIX is not set or is not a directory.  Please make sure it's set globally or in conf/accumulo-env.sh"
-   exit 1
-fi
-if [ -z "${ZOOKEEPER_HOME}" -o ! -d "${ZOOKEEPER_HOME}" ]; then
-   echo "ZOOKEEPER_HOME is not set or is not a directory.  Please make sure it's set globally or in conf/accumulo-env.sh"
-   exit 1
-fi
-
-# This is default for hadoop 2.x;
-#   for another distribution, specify (DY)LD_LIBRARY_PATH
-#   explicitly in ${ACCUMULO_HOME}/conf/accumulo-env.sh
-#   usually something like:
-#     ${HADOOP_PREFIX}/lib/native/${PLATFORM}
-if [ -e "${HADOOP_PREFIX}/lib/native/libhadoop.so" ]; then
-   LIB_PATH="${HADOOP_PREFIX}/lib/native"
-   LD_LIBRARY_PATH="${LIB_PATH}:${LD_LIBRARY_PATH}"     # For Linux
-   DYLD_LIBRARY_PATH="${LIB_PATH}:${DYLD_LIBRARY_PATH}" # For Mac
-fi
-
-# Export the variables just in case they are not exported
-# This makes them available to java
-export JAVA_HOME HADOOP_PREFIX ZOOKEEPER_HOME LD_LIBRARY_PATH DYLD_LIBRARY_PATH
-
-# Strip the instance from $1
-APP=$1
-# Avoid setting an instance unless it's necessary to ensure consistency in filenames
-INSTANCE=""
-# Avoid setting a pointless system property
-INSTANCE_OPT=""
-if [[ "$1" =~ ^tserver-[1-9][0-9]*$ ]]; then
-  APP="$(echo "$1" | cut -d'-' -f1)"
-  # Appending the trailing underscore to make single-tserver deploys look how they did
-  INSTANCE="$(echo "$1" | cut -d'-' -f2)_"
-
-  #Rewrite the input arguments
-  set -- "$APP" "${@:2}"
-
-  # The extra system property we'll pass to the java cmd
-  INSTANCE_OPT="-Daccumulo.service.instance=${INSTANCE}"
-fi
+function main() {
+  scripts=${bin}/../scripts
+
+  if [[ -z $1 ]]; then
+    invalid_args "<command> cannot be empty"
+  fi
+
+  case "$1" in
+    admin|classpath|create-token|gc|init|info|jar|login-info|master|monitor|shell|tserver|tracer|version)
+      exec_java $1 ${*:2}
+      ;;
+    exec)
+      exec_java ${*:2}
+      ;;
+    bootstrap-config)
+      "${scripts}/bootstrap-config.sh" ${*:2}
+      ;;
+    build-native-lib)
+      "${scripts}/build-native-library.sh"
+      ;;
```

Comment:
I was thinking that scripts in the 'bin' directory and their commands should be considered public API while any script in the 'scripts' directory should not. 

For the generate_monitor_certficate and tool.sh scripts, I wasn't sure if they should be considered public API so they are just in scripts dir with no reference from bin.  Do you think they should be considered public API?  If so, I can add them to the `accumulo` command.